docs/src/osdk/reference/manifest.md

```diff
@@ -32,58 +32,103 @@ Below, you will find a comprehensive version of
 the available configuration options in the manifest.
 
 ```toml
-kcmd_args = ["init=/bin/busybox", "path=/usr/local/bin"] # <1>
-init_args = ["sh", "-l"] # <2>
-initramfs="./build/initramfs.cpio.gz" # <3>
-
-[boot]
-loader = "grub" # <4>
-protocol = "multiboot2" # <5>
-grub-mkrescue = "/usr/bin/grub-mkrescue" # <6>
-ovmf = "/usr/bin/ovmf" # <7>
-
-[qemu]
-path = "/usr/bin/qemu-system-x86_64" # <8>
-machine = "q35" # <9>
-args = [ # <10>
-    "-enable-kvm",
-    "-m 2G", 
-    "-device virtio-keyboard-pci,disable-legacy=on,disable-modern=off"
-] 
+[project] # <1>
+type = "kernel"
+
+[run] # <2>
+kcmd_args = ["init=/bin/busybox", "path=/usr/local/bin"] # <3>
+init_args = ["sh", "-l"] # <4>
+initramfs="./build/initramfs.cpio.gz" # <5>
+bootloader = "grub" # <6>
+boot_protocol = "multiboot2" # <7>
+grub-mkrescue = "/usr/bin/grub-mkrescue" # <8>
+ovmf = "/usr/bin/ovmf" # <9>
+qemu_path = "/usr/bin/qemu-system-x86_64" # <10>
+qemu_args = [ # <11>
+    "-machine q35,kernel-irqchip=split",
+    "-cpu Icelake-Server,+x2apic",
+    "-m 2G",
+]
+
+[test] # <2>
+bootloader = "qemu"
+qemu_args = [ # <10>
+    "-machine q35,kernel-irqchip=split",
+    "-cpu Icelake-Server,+x2apic",
+    "-m 2G",
```

Comment:
I am wondering if it is necessary to introduce some kind of template format for the QEMU options. The QEMU options look too complex to be embedded insided Toml.

Reply:
`cargo osdk new` defaults them.

Maybe we can discuss about the Toml macro. Which we cannot get a reference since Cargo does not need it.

```toml
[strings]
QEMU_MACHINE = "-machine q35,kernel-irqchip=split"
[run.qemu]
args = [
    "$QEMU_MACHINE"
# ...
```
Also, selecting characters for eval and escape is important.

But I think it's beyond scope for this PR. :>